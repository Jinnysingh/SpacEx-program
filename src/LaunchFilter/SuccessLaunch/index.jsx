import React from "react";

function SuccessLaunch(props) {
  return (
      <div className="launch-sec">
      <div>
        <u>Successful Launch</u>
      </div>
      {props.launchFilters.map((items) => (
        <button type="button" className="btn btn-success text-dark launch-btn active-btn" onClick={(event) => props.handleFilters(event, "Launch Success")}>
          {items}
        </button>
      ))}
    </div>
  );
}

export default SuccessLaunch;
