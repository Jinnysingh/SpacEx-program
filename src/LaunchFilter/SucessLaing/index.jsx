import React from "react";

function SuccessLaing(props) {
  return (
      <div className="launch-sec">
      <div>
      <u>Successful Laing</u>
      </div>
      {props.launchFilters.map((items) => (
        <button type="button" className="btn btn-success text-dark launch-btn active-btn" onClick={(event) => props.handleFilters(event, "Land Success")}>
          {items}
        </button>
      ))}
    </div>
  );
}

export default SuccessLaing;
