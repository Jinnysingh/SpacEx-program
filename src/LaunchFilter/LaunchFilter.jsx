import React from 'react';
import LaunchYear from './LaunchYears/LaunchYears.jsx';
import SuccessLaunch from './SuccessLaunch/SuccessfulLaunch.jsx';
import SuccessLaing from './SucessLaing/SuccessLaing.jsx';


const launchYears = [2006, 2007, 2008, 2009, 2010, 2011, 2012, 2013, 2014, 2015, 2016, 2017, 2018, 2019, 2020]
const successfulLaunch = ["True", "False"]
const successfulLaing = ["True", "False"]


function LaunchFilter(props) {
    return (
      <aside className="card launch-card mt-20">
      <h5 className="text-left">Filters</h5>
      <LaunchYear launchFilters={launchYears} handleFilters={props.handleFilters}/>
      <SuccessLaunch launchFilters={successfulLaunch} handleFilters={props.handleFilters}/>
      <SuccessLaing launchFilters={successfulLaing} handleFilters={props.handleFilters}/>
      </aside>
    );
  }

  
export default LaunchFilter