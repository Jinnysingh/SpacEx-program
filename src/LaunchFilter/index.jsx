import React from 'react';
import LaunchYear from './LaunchYears/index.jsx';
import SuccessLaunch from './SuccessLaunch/index.jsx';
import SuccessLaing from './SucessLaing/index.jsx';


const launchYears = [2006, 2007, 2008, 2009, 2010, 2011, 2012, 2013, 2014, 2015, 2016, 2017, 2018, 2019, 2020]
const successfulLaunch = ["True", "False"]
const successfulLaing = ["True", "False"]


function LaunchFilter(props) {
    return (
      <aside className="card launch-card mt-20">
      <h5 className="text-left">Filters</h5>
      <LaunchYear launchFilters={launchYears} handleFilters={props.handleFilters}/>
      <SuccessLaunch launchFilters={successfulLaunch} handleFilters={props.handleFilters}/>
      <SuccessLaing launchFilters={successfulLaing} handleFilters={props.handleFilters}/>
      </aside>
    );
  }

  
export default LaunchFilter