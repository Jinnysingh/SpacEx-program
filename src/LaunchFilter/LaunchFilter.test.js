import {shallow} from "enzyme"
import toJSON from "enzyme-to-json"
import React from "react"
import LaunchFilter from "./LaunchFilter.jsx"


test("testing the filters", ()=>{
    const wrapper = shallow(<LaunchFilter />)
    expect(toJSON(wrapper)).toMatchSnapshot();
})