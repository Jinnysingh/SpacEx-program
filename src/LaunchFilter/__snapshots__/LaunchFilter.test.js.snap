// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing the filters 1`] = `
<aside
  className="card launch-card mt-20"
>
  <h5
    className="text-left"
  >
    Filters
  </h5>
  <LaunchYear
    launchFilters={
      Array [
        2006,
        2007,
        2008,
        2009,
        2010,
        2011,
        2012,
        2013,
        2014,
        2015,
        2016,
        2017,
        2018,
        2019,
        2020,
      ]
    }
  />
  <SuccessLaunch
    launchFilters={
      Array [
        "True",
        "False",
      ]
    }
  />
  <SuccessLaing
    launchFilters={
      Array [
        "True",
        "False",
      ]
    }
  />
</aside>
`;
